app/(5-chatbot)/chat/weather.tsx

```diff
@@ -79,6 +85,19 @@ export default function Weather({
 
   return (
     <div className="text-white p-8 rounded-3xl backdrop-blur-lg bg-gradient-to-b from-blue-400 to-blue-600 shadow-lg">
+      <button
+        disabled={clicked}
+        onClick={async () => {
+          setClicked(true);
+          append({
+            parts: [{ text: "Get weather in a random place", type: "text" }],
+            role: "user",
+            id: generateId(),
```

Comment:
in testing, if I leave the generateId call out it generates an id in the background